### DISCUSSION TITLE
how to achieve graphics acceleration without X11 xf86-video-imx-vivante?

### DISCUSSION THREAD

**Discussion by vishnumotghare-eaton** [Freescale/meta-freescale#1712] [Q&A] [answered]
Hello,

I have an IMX6 solo-based custom board and the following are my build configuration.

```
Yocto:  Kirkstone

Kernel:  5.15 (Mainline)

GPU Driver:   Etnaviv

X Driver: xf86-video-armada (ethnaviv enabled)

OpenGL: OpenGL ES 1.1
```

When I run the custom-made benchmark application I see a 50% drop in FPS. The same application on the board with WInCE7 gives better performance.

Just to compare the performance with the IMX Vivante driver, I want to try with following settings 

```
Kernel:   5.15-2.1.x-imx (IMX)

GPU Driver: Vinante

X Driver: xf86-video-imx-vivante

OpenGL: OpenGL ES 1.1
```
But I see the following commit has removed the X11 xf86-video-imx-vivante driver support from imx-gpu-viv
```
commit 9d4e53017385d548aaeec4d8e756ea263c26dc33
Author: Max Krummenacher <max.krummenacher@toradex.com>
Date:   Mon Feb 21 14:53:57 2022 +0100

    xf86-video-imx-vivante: drop vivante specific x11 driver

    The required x11 driver stuff has been removed from imx-gpu-viv and
    thus the X11 driver is no longer buildable.
    Drop the recipe and fallback on the default X11 fb driver.

    | ERROR: Nothing RPROVIDES 'libvivante-dri-imx' (but .../meta-freescale/recipes-graphics/xorg-driver/xf86-video-imx-vivante_6.4.0.p0.0.bb RDEPENDS on or otherwise requires it)

    Completes 747c96f5 ("imx-gpu-viv: Upgrade to 6.4.3.p2.4").

    Signed-off-by: Max Krummenacher <max.krummenacher@toradex.com>

D       recipes-graphics/xorg-driver/xf86-video-imx-vivante/rc.autohdmi
D       recipes-graphics/xorg-driver/xf86-video-imx-vivante_6.4.0.p0.0.bb

```

It is mentioned that it should use the X11 FB driver as a fallback.  The question is how to achieve graphics acceleration. Can the X11 FB driver help to achieve that?

**Comment by MaxKrummenacher** [comment]
Hi

With the "imx-gpu-viv: Upgrade to 6.4.3.p2.4" upgrade NXP dropped X11 support from the proprietary Vivante driver. So what you want to do is unfortunately no longer possible.
The X11 FB driver will not use the GPU for any operation, so no, it cannot help you here.

Max

**Reply by vishnumotghare-eaton** [reply]
Thanks for your reply. I downgraded to version `imx-linux-hardknott` where I see `xf86-video-imx-vivante` package available . I've selected imx6qdlsabresd machine
```
MACHINE=imx6qdlsabresd DISTRO=fslc-x11 source setup-environment build
```
When I try to build `bitbake xf86-video-imx-vivante` I get error like. 
```
imx-gpu-viv was skipped: incompatible with machine imx6qdlsabresd (not in COMPATIBLE_MACHINE)
imx-gpu-viv was skipped: The kernel of machine needs to have Vivante kernel driver support for this recipe to be used.
ERROR: Required build target 'xf86-video-imx-vivante' has no buildable providers.
Missing or unbuildable dependency chain was: ['xf86-video-imx-vivante', 'imx-gpu-viv']
```

I have added machine in COMPATIBLE_MACHINE but It did not fix the error. I was expecting it should build as I'm using imx bsp.   Could you provide some reference for how to fix this error?

Thanks For your Help,
Vishnu

**Reply by MaxKrummenacher** [reply, marked as answer]
You likely didn't configure that you want to build the downstream BSP

E.g. in conf/local.conf:

IMX_DEFAULT_BSP:imx6qdlsabresd = "nxp"

**Reply by vishnumotghare-eaton** [reply]
Thanks. It worked.
